staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit_init.go

```diff
@@ -49,6 +49,10 @@ func withAuditInit(handler http.Handler, newAuditIDFunc func() string) http.Hand
 		// Note: we save the user specified value of the Audit-ID header as is, no truncation is performed.
 		audit.WithAuditID(ctx, types.UID(auditID))
 
+		logger := klog.FromContext(ctx).WithValues("auditID", auditID)
+		ctx = klog.NewContext(ctx, logger)
+		r = r.WithContext(ctx)
```

Comment:
I am working on making this step unnecessary in https://github.com/go-logr/logr/pull/158

The exact solution is still unknown, but it should become faster than modifying the logger and adding it to the context.

Do you have benchmarks that could be run to measure the performance impact of these changes?